pkg/nodebootstrap/assets/bootstrap.al2.sh

```diff
@@ -27,6 +27,14 @@ AWS_SERVICES_DOMAIN="$(get_metadata services/domain)"
 
 source /etc/eksctl/kubelet.env # this can override MAX_PODS
 
+INSTANCE_LIFECYCLE="$(get_metadata instance-life-cycle | grep 'spot')"
```

Comment:
I am curious, why the grep?

I think this would be simpler if we:
- assigned `"$(get_metadata instance-life-cycle)"` to INSTANCE_LIFECYCLE
- added its value directly to the `NODE_LABELS` if it is not empty

So I would remove the grep and only check that it's not empty before adding it to the labels. Although I doubt that it will produce an empty string :thinking:

Reply:
the endpoint `instance-life-cycle` is available only on spot instances. For on demand instances the endpoint doesn't exist - will return a 404

Reply:
The endpoint does exist for all instances :) And it returns `on-demand`:

```
$ TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"` \
> && curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/instance-life-cycle
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    56  100    56    0     0  56000      0 --:--:-- --:--:-- --:--:-- 56000
*   Trying 169.254.169.254...
* TCP_NODELAY set
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /latest/meta-data/instance-life-cycle HTTP/1.1
> Host: 169.254.169.254
> User-Agent: curl/7.61.1
> Accept: */*
> X-aws-ec2-metadata-token: AQAEAITQryp2rkEswx7rHzA4nAXSk3DsOaKVl2YNWfhAlNUejvsK0w==
> 
< HTTP/1.1 200 OK
< X-Aws-Ec2-Metadata-Token-Ttl-Seconds: 21600
< Content-Type: text/plain
< Accept-Ranges: none
< Last-Modified: Mon, 07 Sep 2020 05:28:37 GMT
< Content-Length: 9
< Date: Mon, 07 Sep 2020 06:02:05 GMT
< Server: EC2ws
< Connection: close
< 
* Closing connection 0
on-demand
```

Reply:
You're right. This was added to on demand instances also beginning `IMDSv2`. Made the changes